deploy/buildtools/buildconfig.py

```diff
@@ -65,19 +67,23 @@ def __init__(self,
 
         # run platform specific options
         self.PLATFORM_CONFIG = recipe_config_dict['PLATFORM_CONFIG']
-        self.s3_bucketname = recipe_config_dict['s3_bucket_name']
-        if valid_aws_configure_creds():
-            aws_resource_names_dict = aws_resource_names()
-            if aws_resource_names_dict['s3bucketname'] is not None:
-                # in tutorial mode, special s3 bucket name
-                self.s3_bucketname = aws_resource_names_dict['s3bucketname']
         self.post_build_hook = recipe_config_dict['post_build_hook']
 
-        fpga_bit_builder_dispatcher_class_name = recipe_config_dict.get('fpga-platform', "F1BitBuilder")
-        # create run platform dispatcher object using class given and pass args to it
-        self.fpga_bit_builder_dispatcher = getattr(
-            import_module("buildtools.bitbuilder"),
-            fpga_bit_builder_dispatcher_class_name)(self)
+        # retrieve the bitbuilder section
+        bitbuilder_config_file = None
+        with open(bitbuilderconfigfile, "r") as yaml_file:
+            bitbuilder_config_file = yaml.safe_load(yaml_file)
+
+        bitbuilder_name = recipe_config_dict.get('bit_builder', "f1_bit_builder")
```

Comment:
If we don't want any hidden defaults, we can make this `dict[...]` and put the KV in the recipes.